src/SFML/Window/DRM/InputImplUDev.cpp

```diff
@@ -0,0 +1,729 @@
+void InputImpl::setMousePosition(const Vector2i& position)
+{
+    Lock lock(inpMutex);
+    mousePos = position;
+}
+
+
+////////////////////////////////////////////////////////////
+void InputImpl::setMousePosition(const Vector2i& position, const WindowBase& /*relativeTo*/)
+{
+    setMousePosition(position);
+}
+
+
+////////////////////////////////////////////////////////////
+bool InputImpl::isTouchDown(unsigned int finger)
+{
+    for (std::vector<TouchSlot>::iterator slot = touchSlots.begin(); slot != touchSlots.end(); ++slot)
+    {
+        if (slot->id == static_cast<int>(finger))
+            return true;
+    }
+
+    return false;
+}
+
+
+////////////////////////////////////////////////////////////
+Vector2i InputImpl::getTouchPosition(unsigned int finger)
+{
+    for (std::vector<TouchSlot>::iterator slot = touchSlots.begin(); slot != touchSlots.end(); ++slot)
+    {
+        if (slot->id == static_cast<int>(finger))
+            return slot->pos;
+    }
+
+    return Vector2i();
+}
+
+
+////////////////////////////////////////////////////////////
+Vector2i InputImpl::getTouchPosition(unsigned int finger, const WindowBase& /*relativeTo*/)
+{
+    return getTouchPosition(finger);
+}
+
+
+////////////////////////////////////////////////////////////
+bool InputImpl::checkEvent(sf::Event& ev)
+{
+    Lock lock(inpMutex);
+    if (!eventQueue.empty())
+    {
+        ev = eventQueue.front();
+        eventQueue.pop();
+
+        return true;
+    }
+
+    if (eventProcess(ev))
+    {
+        return true;
+    }
+    else
+    {
+        // In the case of multitouch, eventProcess() could have returned false
+        // but added events directly to the queue.  (This is ugly, but I'm not
+        // sure of a good way to handle generating multiple events at once.)
+        if (!eventQueue.empty())
+        {
+            ev = eventQueue.front();
+            eventQueue.pop();
+
+            return true;
+        }
+    }
+
+    return false;
+}
+
+
+////////////////////////////////////////////////////////////
+void InputImpl::setTerminalConfig()
+{
+    sf::Lock lock(inpMutex);
+    init();
+
+    tcgetattr(STDIN_FILENO, &newt);          // get current terminal config
+    oldt = newt;                             // create a backup
+    newt.c_lflag &= ~(tcflag_t)ECHO;         // disable console feedback
+    newt.c_lflag &= ~(tcflag_t)ISIG;         // disable signals
+    newt.c_lflag |= ICANON;                  // disable noncanonical mode
+    newt.c_iflag |= IGNCR;                   // ignore carriage return
+    tcsetattr(STDIN_FILENO, TCSANOW, &newt); // set our new config
+    tcflush(STDIN_FILENO, TCIFLUSH);         // flush the buffer
+}
+
+
+////////////////////////////////////////////////////////////
+void InputImpl::restoreTerminalConfig()
+{
+    sf::Lock lock(inpMutex);
+    init();
+
+    tcsetattr(STDIN_FILENO, TCSANOW, &oldt); // restore terminal config
+    tcflush(STDIN_FILENO, TCIFLUSH);         // flush the buffer
+}
```

Comment:
What exactly are these two functions achieving? Is this just a debugging feature or something special to DRM?

Reply:
I kind of understood that from the comment, but why do we have to store and restore terminal configs?